src/strands_tools/use_browser.py

```diff
@@ -0,0 +1,1010 @@
+import asyncio
+import json
+
+# Configure logging
+import logging
+import os
+import time  # Added for timestamp in screenshot filenames
+from typing import Dict, List, Optional
+
+import nest_asyncio
+from playwright.async_api import (
+    Browser,
+    BrowserContext,
+    Page,
+    Playwright,
+    async_playwright,
+)
+from playwright.async_api import (
+    Error as PlaywrightError,
+)
+from playwright.async_api import (
+    TimeoutError as PlaywrightTimeoutError,
+)
+from rich.console import Console
+from rich.panel import Panel
+from rich.text import Text
+from strands import tool
+
+from strands_tools.utils.user_input import get_user_input
+
+# Only configure this module's logger, not the root logger
+enable_debug = os.getenv("ENABLE_DEBUG_BROWSER_LOGS", "false").lower() == "true"
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG if enable_debug else logging.INFO)
+if not logger.handlers:
+    handler = logging.StreamHandler()
+    handler.setFormatter(logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s"))
+    logger.addHandler(handler)
+logger.propagate = False
+
+console = Console()
+
+# Global browser manager instance
+_playwright_manager = None
+
+# Apply nested event loop support
+nest_asyncio.apply()
+
+
+# Browser manager class for handling browser interactions
+class BrowserManager:
+    def __init__(self):
+        self._playwright: Optional[Playwright] = None
+        self._browser: Optional[Browser] = None
+        self._context: Optional[BrowserContext] = None
+        self._page: Optional[Page] = None
+        self._cdp_client = None
+        self._user_data_dir = None
+        self._profile_name = None
+        self._tabs = {}  # Dictionary to track tabs by ID
+        self._active_tab_id = None  # Currently active tab ID
+        self._loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self._loop)
+        self.action_configs = {
+            "navigate": {
```

Comment:
Rather than config as code, would it make sense to go to a defined object?

```
class ApiMethods:
    def navigate(page: Page, url: str):
        page.wait_for_load_state("networkidle"))
        return "Navigated to {url}"
    ....
```

I think you'd be able to use reflection at runtime to build up the same data here, but it'd be a more declarative way of exposing actions